R-package/R/lightgbm.R

```diff
@@ -108,14 +108,6 @@ NULL
 #'                            say "the first and tenth columns").}
 #'        \item{\code{reset_data}: Boolean, setting it to TRUE (not the default value) will transform the booster model
 #'                          into a predictor model which frees up memory and the original datasets}
-#'         \item{\code{boosting}: Boosting type. \code{"gbdt"}, \code{"rf"}, \code{"dart"} or \code{"goss"}.}
```

Comment:
I didn't remove `...` in `lightgbm()` because it can be used to pass through keyword arguments to `lgb.train()`.

https://github.com/microsoft/LightGBM/blob/a91e4b2d1877d86030d2de5d07e9257aebc9f143/R-package/R/lightgbm.R#L150-L161

So for this PR, I just removed from the docs any mentions of LightGBM parameters from https://lightgbm.readthedocs.io/en/latest/Parameters.html (since those will now result in "unrecognized argument" errors).